### PR TITLE
Add tutorial on how to make plugins

### PR DESCRIPTION
**Pull request by vnen**
One article with two basic tutorials: how to make a custom node and how to
add a custom dock to the editor.

### CODE REVIEW

**Comment by akien-mga** [comment]
Awesome!

File: tutorials/making_plugins.rst [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+A custom node
+~~~~~~~~~~~~~~~~~~~~
+
+Sometimes you want a certain behavior in many nodes. Maybe a custom scene
+or control that can be reused. Instancing is helpful in a lot of case but
+sometimes it can be cumbersome, especially if you're using it between many
+projects. A good solution to this is to make a plugin that adds a node with a
+custom behavior.
+
+To create a new node type, you can avail of the can avail of the funcion
```
Comment (OvermindDL1) [inline tutorials/making_plugins.rst]: s/funcion/function/  :-)

Comment (vnen) [inline tutorials/making_plugins.rst]: I like how you mentioned the typo but didn't say I wrote "can avail of the" twice :laughing:

Comment (OvermindDL1) [inline tutorials/making_plugins.rst]: Lol, I missed that one the first read-through.  ^.^

**Pull request merged**